### ISSUE TITLE
NamedMapConverter ignores valueType when valueName is null

### ISSUE THREAD

**Issue by briceambrosiak** [x-stream/xstream#42]
Hi there,

I'm trying to marshal some map of values into XML, and so I have a class field of type Map<String,Integer>. I want it marshaled in the following format:

```
<values>
  <val key="A">321</val>
  <val key="B">456</val>
  ...
</values>
```

I've added an annotation on the field like this:

```
@XStreamConverter(MapStringIntegerConverter.class)
private Map<String,Integer> keys;
```

And the class MapStringIntegerConverter extends NamedMapConverter, and looks like this:

```
public class MapStringIntegerConverter extends NamedMapConverter {
    public MapStringIntegerConverter(Mapper mapper, ConverterLookup lookup) {
        super(mapper,
                "key",                  // Name of encapsulating tag
                "name", String.class,   // Store key as attribute "name"
                null, Integer.class,    // Store value as node content
                true,                   // Store key as attributed
                true,                   // Store value as node content
                lookup);
    }
}
```

The problem is that when the XML is loaded, all values are loaded as String instead of Integer.
It works however when I put the value in an attribute or named tag (if I put something instead of "null").

But it should work with 'null', no?

**Comment by joehni** [comment]
This is a duplicate of #36. Give the tip revision of the 1.4.x branch a try (or simply take a copy of its latest NamedMapConverter).

**Comment by briceambrosiak** [comment]
I have the issue with version 1.4.8. Not sure it's a duplicate though, the other bug concerns a canConvert method that I'm not using.

**Comment by joehni** [comment]
As said, this has already been fixed and is part of the 1.4.9 release.
